frontend/lib/resolution/scope-queries.cpp

```diff
@@ -1077,11 +1077,36 @@ const owned<ResolvedVisibilityScope>& resolveVisibilityStmtsQuery(
     result = toOwned(new ResolvedVisibilityScope(scope));
     auto r = result.get();
     // Visit child nodes to find use/import statements therein
+    std::vector<const AstNode*> todo;
+    std::vector<const AstNode*> requireNodes;
```

Comment:
Since there's already an `isRequire` when inserting into this list, why not use `toRequire` and narrow the type to `const Require*`? This way, you don't have to do an (unchecked, though safe) `toRequire` cast later on. Sorry if this is too micromanage-y, feel free to merge without this suggestion.